src/shapes/space_filler/SpaceFiller.js

```diff
@@ -0,0 +1,175 @@
+  constructor() {
+    super('Space Filler')
+    this.linkText = 'Fractal charm: space filling curves'
+    this.link = 'https://www.youtube.com/watch?v=RU0wScIj36o'
+  }
+
+  getInitialState() {
+    return {
+      ...super.getInitialState(),
+      ...{
+        type: 'space_filler',
+        selectGroup: 'Erasers',
+        repeatEnabled: false,
+        canTransform: false,
+        startingSize: 1,
+        canChangeSize: false,
+        fillerOrder: 6,
+        fillerType: 'Hilbert'
+      }
+    }
+  }
+
+  // Algorithm adapted from https://editor.p5js.org/simontiger/full/2CrT12N4
+  getVertices(state) {
+    const machine = state.machine
+    const shape = state.shape
+    const order = shape.fillerOrder || 1
+
+    let sizeX, sizeY
+    if (machine.rectangular) {
+      sizeX = machine.maxX - machine.minX
+      sizeY = machine.maxY - machine.minY
+    } else {
+      sizeX = sizeY = machine.maxRadius * 2.0
+    }
+
+    let config = fillers[shape.fillerType]
+    if (shape.fillerType === 'Morton') {
+      // generate our vertices using space filling curve recursion
+      config.sizeX = sizeX
+      config.sizeY = sizeY
+      return this.spaceFillingCurve(order, config)
+    } else {
+      // generate our vertices using a set of l-system rules
+      let config = fillers[shape.fillerType]
+      config.steps = order
+
+      if (config.side === undefined) { config.side = 5 }
+      if (config.angle === undefined) { config.angle = Math.PI/2 }
+
+      let curve = lsystemPath(lsystem(config), config)
+      let scale = config.orderGrow ? order : 1
+
+      return resizeVertices(curve, sizeX*scale, sizeY*scale)
+    }
+  }
+
+  getOptions() {
+    return options
+  }
+
+  spaceFillingCurve(order, config) {
+    // create initial seed vertices that are sized to fill the entire machine area
+    let size = Math.max(config.sizeX, config.sizeY) / config.grid
+    let seedPoints = []
+    let seedPointsOrdered = []
+
+    for (let j = 0; j < config.grid; j++) {
+      for (let i = 0; i < config.grid; i++) {
+        seedPoints.push(new Victor(i*size + size/2, j*size + size/2))
+      }
+    }
+
+    for (let i=0; i<config.seed.length; i++) {
+      seedPointsOrdered.push(seedPoints[config.seed[i]])
+    }
+
+    config.seedPointsOrdered = seedPointsOrdered
+    config.size = size
+
+    // recursively generate our curve
+    let curve = this.fillSpace(order, config)
+    return curve.map(vertex => vertex.add({x: -size*config.grid/2, y: -size*config.grid/2}))
+  }
+
+  // Algorithm adapted from https://editor.p5js.org/simontiger/full/2CrT12N4
+  // Needed at the moment only for the Morton curve because I could not find
+  // L-system rules to define one.
+  fillSpace(order, config) {
+    const {
+      seed,
+      seedPointsOrdered,
+      grid,
+      rotation,
+      size
+    } = config
+
+    if (order === 1) {
+      return seedPointsOrdered
+    }
+
+    const prevOrder = this.fillSpace(order - 1, config)
```

Comment:
Should we add some kind of an error/escape/emergency stop in case this ends up computing way too long/many loops? Just something to error, and return a half pattern, instead of crashing the tab? I hit a condition with the flow snake where it ran out of memory.

Reply:
Flow snake has a high max, because it looks pretty, but it is computationally intensive. I can reduce the max. Regarding a limit, we may want that at a higher level (total loops or whatever), particularly once we allow multiple layers.